authors/apps/authentication/backends.py

```diff
@@ -1,14 +1,51 @@
-# import jwt
-#
-# from django.conf import settings
-#
-# from rest_framework import authentication, exceptions
-#
-# from .models import User
+import jwt
+from django.conf import settings
+from rest_framework import authentication, exceptions
+from .models import User
 
 """Configure JWT Here"""
 
+
 class JWTAuthentication:
-    def authenticate(self, request):
-        pass
+    """
+    This class implements a custom authentication by overriding
+    the .authenticate(self, request) method. The method returns
+    a two-tuple of (user, token) on successfull authentication
+    and None  otherwise.
+    """
 
+    def authenticate_header(self, request):
+        """
+        This method returns 'None' if authentication is not attempted
+        Otherwise it returns the token.
+        """
+        header = authentication.get_authorization_header(request)
+        if header:
+            token = header.split()[1].decode('utf-8')
+            return token
+        return None
```

Comment:
Raise an Exception if token is None

Reply:
Hey, thanks for the review. However, I still have issues implementing the Exception feedback. The `jwtauthentication` class must either return a `token` to endpoints that are authenticated or `None` to endpoints that do not require authentication. In this logic when I raise an exception, the endpoints that do not require authentication fails. I welcome any suggestion you give me. thanks